app/Http/Controllers/Api/AssetsController.php

```diff
@@ -364,19 +364,29 @@ public function index(Request $request, $audit = null)
     /**
      * Returns JSON with information about an asset (by tag) for detail view.
      *
-     * @author [A. Gianotto] [<snipe@snipe.net>]
      * @param string $tag
+     * @param string $deleted
      * @since [v4.2.1]
-     * @return JsonResponse
+     * @author [A. Gianotto] [<snipe@snipe.net>]
+     * @return \Illuminate\Http\JsonResponse
      */
     public function showByTag(Request $request, $tag)
     {
-        if ($asset = Asset::with('assetstatus')->with('assignedTo')->where('asset_tag', $tag)->first()) {
-            $this->authorize('view', $asset);
+        $this->authorize('index', Asset::class);
```

Comment:
So it looks like we're now authorizing you based on whether you can view the *index* page, rather than view the asset? I worry that with multicompany support that could be a problem. Maybe you should need to have both? 'index' to actually do the query, and the 'view on the asset' to get the results?